encoding/codecv2/codecv2.go

```diff
@@ -341,6 +385,46 @@ func constructBlobPayload(chunks []*encoding.Chunk, useMockTxData bool) (*kzg484
 	return blob, blobVersionedHash, &z, nil
 }
 
+// DecodeTxsFromBlob decodes txs from blob bytes and writes to chunks
+func DecodeTxsFromBlob(blob *kzg4844.Blob, chunks []*DAChunkRawTx) error {
+	compressedBytes := codecv1.BytesFromBlobCanonical(blob)
+	magics := []byte{0x28, 0xb5, 0x2f, 0xfd}
+
+	blobBytes, err := decompressScrollBatchBytes(append(magics, compressedBytes[:]...))
+	if err != nil {
+		return err
+	}
+
+	numChunks := int(binary.BigEndian.Uint16(blobBytes[0:2]))
```

Comment:
could we reuse codecv1's implementation by some refactor? seems like MaxNumChunks is 45 in codecv2, and 15 in codecv1.

I'm good at both styles, if there is no neat refactor.

Reply:
difference not only in NumChunks, codecv2 also uses decompression